tools/RepositoryManagement/DevHome.RepositoryManagement/ViewModels/RepositoryManagementItemViewModel.cs

```diff
@@ -2,61 +2,451 @@
+
+        _log.Error(ex, string.Empty);
+    }
+
+    private async Task CloneLocationNotFoundNotifyUser()
+    {
+        // strings need to be localized
+        var cantFindRepositoryDialog = new ContentDialog()
+        {
+            XamlRoot = _window.Content.XamlRoot,
+            Title = $"Can not find {RepositoryName}.",
+            Content = $"Cannot find {RepositoryName} at {Path.GetFullPath(ClonePath)}.  Do you know where it is?",
+            PrimaryButtonText = $"Locate {RepositoryName} via File Explorer.",
+            SecondaryButtonText = "Remove from list",
+            CloseButtonText = "Cancel",
+        };
+
+        var dialogResult = await cantFindRepositoryDialog.ShowAsync();
+
+        // User will show DevHome where the repository is.
+        // Open the folder picker.
+        // Maybe don't close the dialog until the user is done
+        // with the folder picker.
+        if (dialogResult == ContentDialogResult.Primary)
+        {
+            var newLocation = await PickNewLocationForRepositoryAsync();
+            if (string.IsNullOrEmpty(newLocation))
+            {
+                _log.Information("The path from the folder picker is either null or empty.  Not updating the clone path");
+                return;
+            }
+
+            var repository = GetRepositoryReportIfNull(nameof(CloneLocationNotFoundNotifyUser));
+            if (repository == null)
+            {
+                return;
+            }
+
+            // The repository exists at the location stored in the Database
+            // and the new location is set.
+            var didUpdate = _dataAccess.UpdateCloneLocation(repository, Path.Combine(newLocation, RepositoryName));
+
+            if (!didUpdate)
+            {
+                _log.Warning($"Could not update the database.  Check logs");
+            }
+        }
+        else if (dialogResult == ContentDialogResult.Secondary)
+        {
+            RemoveThisRepositoryFromTheList();
+            return;
+        }
+    }
+
+    private Repository GetRepositoryReportIfNull(string action)
+    {
+        var repository = _dataAccess.GetRepository(RepositoryName, ClonePath);
+
+        // The user clicked on this menu from the repository management page.
+        // The repository should be in the database.
+        // Somehow getting the repository returned null.
+        if (repository is null)
+        {
+            _log.Warning($"The repository with name {RepositoryName} and clone location {ClonePath} is not in the database when it is expected to be there.");
+            TelemetryFactory.Get<ITelemetry>().Log(
+                EventName,
+                LogLevel.Critical,
+                new RepositoryLineItemEvent(action, RepositoryName));
+
+            return null;
+        }
+
+        return repository;
+    }
+
+    private async Task CheckCloneLocationNotifyUserIfNotFound()
```

Comment:
CheckCloneLocation**And**NotifyUserIfNotFound

Reply:
Changed.